src/Tizen.NET.MaterialComponents/Components/MSheets.cs

```diff
@@ -0,0 +1,244 @@
+            set => _panel = value;
+        }
+
+        protected double Ratio
+        {
+            get => _ratio;
+            set => UpdateRatio(value);
+        }
+
+        protected void UpdateDirection(MSheetsDirection direction)
+        {
+            if (direction == MSheetsDirection.Bottom)
+            {
+                Panel.Direction = PanelDirection.Bottom;
+                Ratio = _bottomSheetDefaultRatio;
+            }
+            else
+            {
+                Panel.Direction = PanelDirection.Right;
+                Ratio = _sideSheetDefaultRatio;
+            }
+        }
+
+        void UpdateBackSheet(EvasObject contents)
+        {
+            if (_backSheet != null)
+            {
+                _backContainer.UnPack(_backSheet);
+                _backSheet.Hide();
+            }
+            _backSheet = contents;
+            if (_backSheet != null)
+            {
+                if (!_backSheet.IsVisible)
+                {
+                    _backSheet.Show();
+                }
+                _backContainer.PackEnd(_backSheet);
+
+                UpdateBackSheetGeometry();
+            }
+        }
+
+        void UpdateBackSheetGeometry()
+        {
+            if (_backSheet != null)
+            {
+                _backContainer.Geometry = Geometry;
+            }
+            UpdateFrontSheetGeometry();
+        }
+
+        void UpdateFrontSheet(EvasObject contents)
+        {
+            if (_frontSheet != null)
+            {
+                _frontSheet.Hide();
+            }
+            _frontSheet = contents;
+            if (_frontSheet != null)
+            {
+                _frontSheet.SetAlignment(-1, -1);
+                _frontSheet.SetWeight(1, 1);
+                if (!_frontSheet.IsVisible)
+                {
+                    _frontSheet.Show();
+                }
+                Panel.SetContent(_frontSheet, true);
+
+                UpdateScrim();
+                UpdateFrontSheetGeometry();
+            }
+        }
+
+        void UpdateFrontSheetGeometry()
+        {
+            if (_frontSheet != null)
+            {
+                _scrim.Geometry = Geometry;
+                Panel.Geometry = Geometry;
+            }
+        }
+
+        void Initialize(EvasObject parent, MSheetsDirection direction)
+        {
+            _backContainer = new Box(parent)
+            {
+                AlignmentX = -1,
+                AlignmentY = -1,
+                WeightX = 1,
+                WeightY = 1,
+            };
+            _backContainer.Show();
+            PackEnd(_backContainer);
+
+            _scrim = new Box(parent)
+            {
+                AlignmentX = -1,
+                AlignmentY = -1,
+                WeightX = 1,
+                WeightY = 1,
+                BackgroundColor = MColors.Current.OnSurfaceColor.WithAlpha(0.32)
+            };
+            _scrim.Show();
+            PackEnd(_scrim);
+
+            Panel = new Panel(parent);
```

Comment:
Do you mean `MSheets` only support modal side sheets? is there no support for standard side sheets?

Reply:
Changed to support only modal sheets.